shared/src/store/action-reducers/vehicle.ts

```diff
@@ -139,11 +144,13 @@ export namespace VehicleActionReducers {
                 );
             }
             draftState.vehicles[vehicle.id] = cloneDeepMutable(vehicle);
-            for (const material of materials) {
-                draftState.materials[material.id] = cloneDeepMutable(material);
+            for (const material of cloneDeepMutable(materials)) {
+                draftState.materials[material.id] = material;
+                addElementPosition(draftState, 'materials', material);
```

Comment:
this should be wrong, as material is initially loaded in the vehicle and its `position === undefined`